pygccxml/declarations/type_traits.py

```diff
@@ -297,6 +303,16 @@ def remove_const(type):
     if not is_const(nake_type):
         return type
     else:
+        if isinstance(nake_type, cpptypes.array_t):
+            is_v = is_volatile(nake_type)
+            if is_v:
+                base_type = nake_type.base.base.base
+            else:
+                base_type = nake_type.base.base
+            result_type = base_type
+            if is_v:
+                result_type = cpptypes.volatile_t(result_type)
```

Comment:
Here you are generating new instances, and return them. I think (I did not test this) that this may be a problem if a comparison needs to be done (using is_same() method for example). Could the types be retrieved by returning the right .base ?
The same comment holds for the remove_volatile() method.